src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-event-buffer-time.mdx

```diff
@@ -0,0 +1,582 @@
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$maxBufferTimeInSec`
+      </td>
+
+            <td>
+        `int`
+      </td>
+
+      <td>
+        Required. The maximum time (in seconds) that the agent should store events in memory. The default value harvest cycle length is 600 seconds.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if it succeeds, or `false` if it doesn't.
+
+## Example [#example]
+
+Here's an example to set max event buffer time to 300 seconds:
+
+### Java [#java]
+
+```java
+boolean fiveMinuteLimitSet = NewRelic.setMaxEventBufferTime(300);
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
+
+        </TabsPageItem>
+        <TabsPageItem id="ios">
+
+## Syntax [#syntax]
+
+### Objective-c [#objc]
+
+
+```objectivec
++ (void) setMaxEventBufferTime:(unsigned int)seconds;
+```
+
+### Swift [#swift]
+
+
+```swift
+NewRelic.setMaxEventBufferTime:(unsigned int)seconds;
+```
+
+## Description [#description]
+
+Sets the event harvest cycle length. Default is 600 seconds.
+
+This method sets the maximum time in seconds that the New Relic iOS agent will store events in memory before including the buffered event to be sent to New Relic in the next [harvest cycle](/docs/accounts-partnerships/education/getting-started-new-relic/glossary#harvest-cycle).
+In other words, when the oldest event timestamp exceeds this custom configured time, the agent transmits the buffered content at the harvest cycle between each connection from a New Relic agent to the collector. You must set this value after `Agent.start()` is called.
+
+* Default is 600 seconds (10 minutes).
+* Minimum value cannot be less than 60 seconds.
+
+See also [`setMaxEventPoolSize()`](/docs/mobile-monitoring/new-relic-mobile-ios/ios-sdk-api/setmaxeventpoolsize-ios-sdk-api), which lets you change the maximum size of the event pool.
+
+For context on how to use this API, see the documentation about sending custom attributes and events:
+
+* [Objective-C](/docs/mobile-monitoring/new-relic-mobile-ios/install-configure/work-ios-sdk-api#objc-custom-att-events)
+* [Swift](/docs/mobile-monitoring/new-relic-mobile-ios/install-configure/work-ios-sdk-api#swift-custom-att-events)
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+
+      <th width={200}>
+        Type
+      </th> 
+
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `seconds`
+      </td>
+
+      <td>
+        `unsigned int`
+      </td>
+
+      <td>
+        Required. The maximum amount of time in seconds that events can be buffered before they are sent to New Relic.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if it succeeds, or `false` if it doesn't.
```

Comment:
@mbruin-NR does this method returns value?

Reply:
It doesn't have a return value.